src/site/asn2.rst

```diff
@@ -98,6 +98,76 @@ Part 5 --- Testing
 Code Peculiarities of Note
 ==========================
 
+Exceptions
+----------
+
+* There are two exceptions included that are not provided by Java by default
+    * ``LocationNotInMazeException``
+    * ``MazeEndpointsException``
+
+* These are used/thrown by the ``Maze`` class under certain circumstances
+    * When a given cell coordinate is not in the maze
+    * When there are issues with the start and/or end cells
+
+
+Final Class and Fields
+----------------------
+
+* You will notice that the ``Maze`` and ``Cell`` classes are set to ``final``
+    * e.g. ``public final class Cell``
+    * All it means is that this class cannot be inherited from
+    * We have yet to discuss inheritance, so do worry too much about this
+
+* The ``Maze`` and ``Cell`` objects' fields are set to ``final``
+    * This means we do not allow any fields to change
+        * You will notice no setters in these classes
+    * We can have Java explicitly make sure they stay unchanged
+    * Before, we only really saw static constants set to final
+
+
+Missing Constructors
+--------------------
+
+* In the ``MazeRenderer`` and ``DfsMazeSolver``, you will notice there are no constructors
+* This may seem strange, but if you do not write a constructor, Java is still happy to create an instance of the object
+    * e.g. ``MazeRenderer renderer = new MazeRenderer();``
```

Comment:
Java will add the default constructor (the no-params one) for you if no constructor exists.